controllers/DataSetAdmin.js

```diff
@@ -0,0 +1,271 @@
+      req.flash('error', "Error processing rows")
+      return res.redirect(this.routePrefix)
+    })
+  }
+
+  /**
+   * Transform the datarow after storage,
+   * associating with the given dataset ID.
+   * @param  {[datarow]} row       DataRow instance to put under given DataSet ID
+   * @param  {[string]} dataSetId   Parent DataSet ID
+   * @return the original row passed in, with modifications.
+   */
+  _prepareDataRowInDataSet(row, dataSetId) {
+    _.each(row, (val,key) => {
+      if (typeof(val) == 'string') {
+        if (val.endsWith('%')) {
+          let what = Number.parseFloat(val)
+          if (_.isNumber(what) && !Number.isNaN(what)) {
+            val = what
+          }
+        }
+      } else if (typeof(val) == 'undefined') {
+        val = null
+      }
+      row[key] = val
+    })
+    row.dataset = dataSetId
+    //nb: must return changed row rather than new object
+    return row
+  }
+
+  /**
+   * Based on supplied data row, build array of field info objects
+   * which include a unique field-id for each property on the row.
+   * @param  {[datarow]} sample [representative row of data]
+   * @return { array of object}        [name, id, and type properties for each field]
+   */
+  _buildFieldInfo(sample) {
+    let fields = new Array()
+    _.each(_.keys(sample), (key,index) => {
+      let type = field.STRING_TYPE
+      let val = sample[key]
+      if (typeof(val) == 'string') {
+        if (val.endsWith('%')) {
+          type = field.PERCENT_TYPE
+        } else if (! isNaN(parseFloat(val))) {
+          type = field.DECIMAL_TYPE
+        }
+      }
+      if ( type != field.PERCENT_TYPE && 
+        ((-1 < key.toLowerCase().indexOf('percent')) || (-1 < key.indexOf('%'))) ) {
+        type = field.PERCENT_TYPE
+      }
+      if (! _.contains(["true", "id", "dataset", "createdAt", "updatedAt"], key)) {
+        let field = field.FIELD_DEFAULTS
+        field.name = key
+        field.id = this._generateUniqueId()
+        field.label = morph.toTitle(key)
+        field.type = type
+        fields.push(field)
+      }
+    })
+    return fields
+  }
+
+   /**
+    * override EditController save:
+   * handle POST response for edit existing 
+   * or create new dataset
+   */
+  save(req, res) {
+    let dataSetId = req.params.id
+    this.log.debug("save DataSet id: ", dataSetId)
+    if (dataSetId) {
+      this.model.findOne(dataSetId)
+      .then((dataset) => {
+        dataset.name = req.body.name
+        dataset.source = req.body.source
+        dataset.description = req.body.description
+        _.each( dataset.fields, (dsField) => {
+          dsField.visible = false;
+          dsField.primaryKey = false;
+        })
+        _.each( req.body, (value,key) => {
+          //update fields from request parameters field_<field.id>_<key>
+          if (0 <= key.indexOf('field_')) {
+            this.log.debug( "save req body key ", key, " value ", value)
+            let pComponents = key.split('_')

```

Comment:
These name conventions (field_whatever_? and isSomething), are they coordinated by something on the client side I'm not seeing here or are they just specific to WPSP's original data?